app/controllers/api/v1/links_controller.rb

```diff
@@ -2,7 +2,11 @@ class Api::V1::LinksController < Api::BaseController
   include LinksHelper
 
   before_action :authenticate_with_api_key
-
+  def show
+    param = { long_url: params[:url] }
+    @link = @user.links.build(param)
```

Comment:
We don't need to build another link with the show action, just display the link!  This line can be removed.